docs/lib/troubleshooting/fragments/js/upgrading.md

```diff
@@ -0,0 +1,154 @@
+When upgrading Amplify packages, it is important to make sure that there are no duplicate versions of Amplify packages in the `node_modules` folder tree as a result of the upgrade. Having multiple versions of packages can yield unexpected behavior as modules imported in your code might point to versions not configured by Amplify when calling `Amplify.configure`. To prevent this situation, you can [Check for duplicate versions](#check-for-duplicate-versions), and if duplicate versions exists, then [Upgrade Amplify packages](#upgrade-amplify-packages).
```

Comment:
Can you include the likely error-scenarios that would lead one to this documentation? For example, "If you're seeing... X, Y, Z, then you may want to check the following..."?

Reply:
Once aws-amplify/amplify-js#8083 is merged, customers most likely will arrive here coming from the [bug issue template checklist](https://github.com/aws-amplify/amplify-js/pull/8083/files#diff-7e6dd26cf81256b658eb44be8a5664393c3d886d8de97606d7f89f691d98b48d) or the [messages logged by the library](https://github.com/aws-amplify/amplify-js/pull/8083/files#diff-db69e785d7fc5c73340d7a6b0eb17b7804601ee6143bb4d95df2968ace533c80)

Would you like to see the scenarios being mentioned here too?

Reply:
I would. We can't anticipate how customers will arrive at a given documentation resource. Theoretically, folks could arrive via search. Also, we're including the resource in our top-level navigation. In other words, "Every page is page one." We should provide more context, here.